rootless_linux.go

```diff
@@ -28,6 +30,23 @@ func shouldUseRootlessCgroupManager(context *cli.Context) (bool, error) {
 		return false, nil
 	}
 	// euid = 0, in a userns.
+	//
+	// [systemd driver]
+	// We can call DetectUID() to parse the OwnerUID value from `busctl --user --no-pager status` result.
+	// The value corresponds to sd_bus_creds_get_owner_uid(3).
+	// If the value is 0, we have rootful systemd inside userns, so we do not need the rootless cgroup manager.
+	//
+	// On error, we assume we are root. An error may happen during shelling out to `busctl` CLI,
+	// mostly when $DBUS_SESSION_BUS_ADDRESS is unset.
+	if context.GlobalBool("systemd-cgroup") {
+		ownerUID, err := systemd.DetectUID()
+		if err != nil {
+			logrus.WithError(err).Debug("failed to get the OwnerUID value, assuming the value to be 0")
```

Comment:
Maybe upgrade this from Debug to a Warning? Otherwise the user will never know why it's not working as rootless.

Reply:
Intentionally demoted to Warning: https://github.com/opencontainers/runc/pull/2725#issuecomment-756997560